src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -173,6 +192,31 @@ def _try_fetch(self) -> list[GlobalInitScriptInfo]:
             yield GlobalInitScriptInfo(*row)
 
 
+class AWSInstanceProfileCrawler(CrawlerBase):
+    """Crawl AWS workspace for Instance profiles"""
+
+    def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
+        super().__init__(sbe, "hive_metastore", schema, "aws_instance_profiles", AWSInstanceProfileInfo)
+        self._ws = ws
+
+    def _crawl(self) -> list[AWSInstanceProfileInfo]:
+        instance_profiles = self._ws.instance_profiles.list()
```

Comment:
```suggestion
        return self._ws.instance_profiles.list()
```
I wonder if we can just do this.

Reply:
Considered that!
Following the Azure example, a new class was created. The new data class lets us mark fields as NULLable which lets us get through the null record filter and create a derived field `account_id` like @pohlposition might be asking for.